amethyst_network/src/metrics/csv_metrics.rs

```diff
@@ -0,0 +1,39 @@
+use super::{MetricObserver, NetworkMetricObject};
+use serde::Serialize;
+use std::fs::File;
+use std::time::{Duration, Instant};
+
+/// This will handle changing metrics and write them to a CSV file.
+/// CSV write interval could be managed by passing in a `Duration` at creation time.
+pub struct CsvMetrics {
+    // the file handle to the CSV file
+    csv_file: File,
+    // this will be used for the writing interval to the console.
+    last_send: Instant,
+    // interval rate
+    update_rate: Duration,
```

Comment:
so it seems that this updated logic is quite repetitive, maybe we can move it to a trait?

Reply:
jup was thinking about that. I'll take it a shot.